pkg/compactor/blocks_cleaner_test.go

```diff
@@ -681,6 +686,95 @@ func TestBlocksCleaner_ShouldRemoveBlocksOutsideRetentionPeriod(t *testing.T) {
 	}
 }
 
+func TestBlocksCleaner_ShouldRemovePartialBlocksOutsideDelayPeriod(t *testing.T) {
+	bucketClient, tempDir := mimir_testutil.PrepareFilesystemBucket(t)
+	bucketClient = bucketindex.BucketWithGlobalMarkers(bucketClient)
+
+	ts := func(hours int) int64 {
+		return time.Now().Add(time.Duration(hours)*time.Hour).Unix() * 1000
+	}
+
+	block1 := createTSDBBlock(t, bucketClient, "user-1", ts(-10), ts(-8), 2, nil)
+	block2 := createTSDBBlock(t, bucketClient, "user-1", ts(-8), ts(-6), 2, nil)
+
+	cfg := BlocksCleanerConfig{
+		DeletionDelay:           time.Hour,
+		CleanupInterval:         time.Minute,
+		CleanupConcurrency:      1,
+		DeleteBlocksConcurrency: 1,
+	}
+
+	ctx := context.Background()
+	logger := test.NewTestingLogger(t)
+	reg := prometheus.NewPedanticRegistry()
+	cfgProvider := newMockConfigProvider()
+
+	cleaner := NewBlocksCleaner(cfg, bucketClient, tsdb.AllUsers, cfgProvider, logger, reg)
+
+	assertBlockExists := func(user string, block ulid.ULID, expectExists bool) {
+		exists, err := bucketClient.Exists(ctx, path.Join(user, block.String(), metadata.MetaFilename))
+		require.NoError(t, err)
+		assert.Equal(t, expectExists, exists)
+	}
+
+	makeBlockPartial := func(user string, block ulid.ULID) {
+		err := os.Remove(path.Join(tempDir, user, block.String(), metadata.MetaFilename))
```

Comment:
Should be `filepath.Join` since it's a filesystem path:
```suggestion
		err := os.Remove(filepath.Join(tempDir, user, block.String(), metadata.MetaFilename))
```

Reply:
alternatively we can delete file via bucket, to avoid mixing bucket and filesystem operations.